bears/vcs/git/GitCommitBear.py

```diff
@@ -269,6 +270,31 @@ def check_body(self, body,
                                'Commit body lines should not exceed {} '
                                'characters.'.format(body_line_length))
 
+    def initialise_issue_ref_regex(self,
+                                   body_close_issue_full_url,
+                                   result_message,
+                                   host
+                                   ):
+        """
+        Helper function to initialize the kind of issue regex to use
+
+        :param body_close_issue_full_url:
+            Sets the regex to either use full body reference or just number
+        :param result_message:
+            The result message having exactly one variable to fill in
+        :param host:
+            The current repository's host
+        """
+        if body_close_issue_full_url:
+            self.ISSUE_TYPE = 'full issue'
+            self.ISSUE_REF_REGEX = (
+                r'https?://{}\S+/issues/(\S+)'.format(re.escape(host)))
+        else:
+            self.ISSUE_TYPE = 'issue'
+            self.ISSUE_REF_REGEX = r'(?:\w+/\w+)?#(\S+)'
```

Comment:
The difference between this and 'hg' URLs, is not actually related to hg at all.

These URLs, and the differences, are a property of the hoster, not the vcs software being used.

see https://github.com/coala/coala-bears/pull/2310#pullrequestreview-116498241